### PR TITLE
Use indexing revision to trigger Blazegraph reindexing

### PR DESCRIPTION
**Pull request by olivergrabinski**
#3569

### CODE REVIEW

**Comment by olivergrabinski** [comment]
To do:
- [ ] Review `indexingRev` name
- [ ] Namespace should match the projection name

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala [hunk 1]
```diff
@@ -45,26 +45,38 @@ final class BlazegraphCoordinator private (
...
+                      supervisor.run(
+                        projection,
+                        for {
+                          _ <- createNamespace(active)
+                          _ <- cache.put(active.ref, active)
+                        } yield ()
+                      )
+                  }
+              case (_, deprecated: DeprecatedViewDef)                                              =>
+                cleanupCurrent(deprecated.ref)
```
Comment (imsdu) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala:77]: I missed it in the ES part but you don't have to do another look up to get the value in the cache

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala [hunk 2]
```diff
@@ -45,26 +45,38 @@ final class BlazegraphCoordinator private (
...
+            (cachedView, v) match {
+              case (Some(cached), active: ActiveViewDef) if cached.projection == active.projection =>
+                for {
+                  _ <- cache.put(active.ref, active)
+                  _ <- Task.delay {
+                         logger.info(s"Index ${active.projection} already exists and will not be recreated.")
+                       }
+                } yield ()
+              case (_, active: ActiveViewDef)                                                      =>
+                IndexingViewDef
```
Comment (imsdu) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala:59]: Maybe create a private method compile here to improve a little readibility ?
Something like:
```scala
private def compile(v: ActiveViewDef) =
    IndexingViewDef.compile(v, compilePipeChain, graphStream, sink(v))
```

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala (hunk 1 above)
Comment (olivergrabinski) [reply delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala]: Corrected in both

**Review by imsdu — APPROVED** [review]

**Pull request merged**